src/os/host-start.r

```diff
@@ -678,3 +628,66 @@ comment [
     return :host-repl
 ]
 
+
+; Define REPL! object for skinning - stub for elsewhere?
+;
+
+repl!: make object! [
+    repl: true
+    counter: 0
+    last-result: _ ;-- last evaluated result (sent by HOST-REPL)
+
+    ; Called on every line of input by HOST-REPL in %os/host-repl.r
+    ;
+    cycle: does [
+        if zero? ++ counter [print-greeting] ;-- only load skin on first cycle
+        counter
+    ]
+
+    ;; APPEARANCE (can be overridden)
+
+    prompt:   {>> }
+    result:   {== }
+    warning:  {!! }
+    error:    {** }                ;; not used yet
+    info:     to-string #{e29398}  ;; info sign!
+    greeting: _
+    print-prompt:   proc []  [print/only prompt]
+    print-result:   proc []  [print unspaced [result last-result]]
+    print-warning:  proc [s] [print unspaced [warning reduce s]]
+    print-error:    proc [e] [print e]
+    print-info:     proc [s] [print [info space space reduce s]]
+    print-greeting: proc []  [boot-print greeting]
+    print-gap:      proc []  [print-newline]
+
+    ;; BEHAVIOR (can be overridden)
+    
+    input-hook: func [
+        {Receives line input, parse/transform, send back to repl eval}
+        s
+    ][
+        s
+    ]
+
+    dialect-hook: func [
+        {Receives code block, parse/transform, send back to repl eval}
+        s
+    ][
+        s
```

Comment:
I don't think I've seen this change before so I couldn't have complained about it :(     But the good news is that i think i like it :)     I do get @earl concerns but I have one suggestion, which I'll mention at end.

So to recap (which may help others because took me a little while to understand).  In the code highlighted above I could have previously stopped a potential function arg screwing things up by changing...

        dialect-hook: func [s] [s]

to

        dialect-hook: func [s [block!]] [s]

Now it won't accept a FUNCTION! arg.

But I don't need that with Ren/C because you cannot pass a FUNCTION! arg unless its explicitly defined in spec:

        >> f: func [a] [type-of a]
        >> f does [1]
        ** Script error: f does not allow function! for its a argument
        ** Where: 
        ** Near: f does [1] ??

        >> f: func [a [function!]] [type-of a
        >> f does [1]
        == integer!

Have i got this right?  If so then this change does make the code a lot safer to a bad function injection.

Looking at @nezdadarek comments - I do find having the option to specify the function spec i'm expecting the arg to have very appealing!  

So to my suggestion to cover @earl concerns.  How about some shortcut in spec like so:

        run-func: func [\f] [f]

Which would expand to...

        run-func: func [f [function!]] [f]

NB.  Doesn't have to be `\` especially as I've already had ideas that this could be used as a `reduce` shortcut.   But example is good for Haskell lovers out there :)